File: Nível 6/2422 - Soma das Casas/main.lua
function encontrar_soma(lista, alvo)
    local numeros = {}

    for i, num in ipairs(lista) do
        local complemento = alvo - num

        if numeros[complemento] then
            return {num, complemento}
        else
            numeros[num] = i
        end
    end
    return nil
end

quantia = tonumber(io.read())
numeros = {}

for i = 1, quantia do 
    num = tonumber(io.read())
    table.insert(numeros, num)
end

alvo = tonumber(io.read())

resultado = encontrar_soma(numeros, alvo)

print(string.format("%d %d", resultado[2], resultado[1]))

File: Nível 6/2422 - Soma das Casas/main.py
def encontrar_soma(lista, alvo):
    numeros = {}  

    for i, num in enumerate(lista):
        complemento = alvo - num 

        if complemento in numeros:
            return [num, complemento]
        else:
            numeros[num] = i  
    return None

quantia = int(input("A: "))
numeros = []
for i in range(1, quantia + 1):
    num = int(input("B: "))
    numeros.append(num)

alvo = int(input("C: "))

resultado = encontrar_soma(numeros, alvo)

print(f"{resultado[1]} {resultado[0]}")